src/main/java/project/myblog/service/PostService.java

```diff
@@ -0,0 +1,64 @@
+package project.myblog.service;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import project.myblog.domain.Member;
+import project.myblog.domain.Post;
+import project.myblog.exception.BusinessException;
+import project.myblog.exception.ExceptionCode;
+import project.myblog.repository.PostRepository;
+import project.myblog.web.dto.post.PostRequest;
+import project.myblog.web.dto.post.PostResponse;
+
+import java.util.Optional;
+
+@Transactional
+@Service
+public class PostService {
+    private final PostRepository postRepository;
+    private final MemberService memberService;
+
+    public PostService(PostRepository postRepository, MemberService memberService) {
+        this.postRepository = postRepository;
+        this.memberService = memberService;
+    }
+
+    public Long createPost(String email, PostRequest requestDto) {
+        Member member = memberService.findMemberByEmail(email);
+        return postRepository.save(requestDto.toEntity(member)).getId();
+    }
+
+    public PostResponse findPost(Long postId) {
+        Post post = findPostById(postId);
+        return new PostResponse(post);
+    }
+
+    public Long updatePost(String email, Long postId, PostRequest requestDto) {
+        Member member = memberService.findMemberByEmail(email);
+
+        Optional<Post> post = postRepository.findById(postId);
+        if (post.isPresent()) {
+            if (post.get().isAuthorization(member)) {
+                return post.get().update(requestDto.getTitle(), requestDto.getContents());
+            }
+            throw new BusinessException(ExceptionCode.POST_AUTHORIZATION);
+        }
+        return postRepository.save(requestDto.toEntity(member)).getId();
```

Comment:
```suggestion
        Post post = findPostById(postId);
        post.update(requestDto.getTitle(), requestDto.getContents(), member);
```

`post`를 수정하는 로직인데 `save`를 할 필요는 없을 것 같으니, `post`를 찾지 못했을 때는 예외를 던져주는게 좋을 것 같습니다.

그리고 내글인지 확인하고 다음 로직을 수행하는 것은 뭔가 절차지향적인 느낌이 드는 것 같은데,
`post.isAuthorization(member)`메서드를 `post.update()`메서드 내부로 가져가보면 어떨까요?

그리고 `updatePost`의 응답이 `204 noContent`이므로 `return`은 없어도 될 것 같아요.